eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1072,11 +1092,70 @@ private ClientResponse makeRemoteCall(Action action) throws Throwable {
      * @throws Throwable
      *             on any error.
      */
-    private ClientResponse makeRemoteCall(Action action, int serviceUrlIndex)
-            throws Throwable {
+    private ClientResponse makeRemoteCall(Action action, int serviceUrlIndex) throws Throwable {
+        String serviceUrl;
+        try {
+            serviceUrl = eurekaServiceUrls.get().get(serviceUrlIndex);
```

Comment:
Validation is done during initialization and there must be at least one service Url.